nvflare/fuel/f3/streaming/byte_streamer.py

```diff
@@ -177,23 +177,13 @@ def _transmit(self, task: TxTask, final=False):
         else:
             payload = wrap_view(task.buffer)[0 : task.buffer_size]
 
-        message = Message(None, payload)
-
-        if task.offset == 0:
-            # User headers are only included in the first chunk
-            if task.headers:
-                message.add_headers(task.headers)
-
-            message.add_headers(
-                {
-                    StreamHeaderKey.CHANNEL: task.channel,
-                    StreamHeaderKey.TOPIC: task.topic,
-                    StreamHeaderKey.SIZE: task.stream.get_size(),
-                }
-            )
+        message = Message(task.headers, payload)
```

Comment:
No, you should keep the original code: message = Message(None, payload), or make a copy of task.headers.
Otherwise you are reusing the headers from task.headers and have the potential of using the headers from previous message.

Reply:
Good catch. Changed.